src/discussion.cairo

```diff
@@ -0,0 +1,109 @@
+#[starknet::interface]
+trait IDiscussion<TContractState> {
+    fn add_comment(ref self: TContractState, prop_id: felt252, ipfs_hash: felt252);
+    fn get_comment(self: @TContractState, prop_id: felt252) -> Array<felt252>;
+} 
+
+#[starknet::component]
+mod discussion {
+    use array::ArrayTrait;
+    use konoha::proposals::IProposals;
+
+    use core::box::Box;
+
+    #[storage]
+    struct Storage {
+        comments: LegacyMap::<(felt252, u64), felt252>,
+        comment_count: LegacyMap::<felt252, u64>
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {}
+
+    #[embeddable_as(DiscussionImpl)]
+    impl Discussions<TContractState, +HasComponent<TContractState>
+    > of super::IDiscussion<ComponentState<TContractState>> {
+        fn add_comment(ref self: ComponentState<TContractState>, prop_id: felt252, ipfs_hash: felt252) {
+            //TODO
```

Comment:
Also check that the submitter is a CARM token holder and save their address to serve alongside the hash. I didn't realise this earlier, sorry for that.